scripts/test/utils

```diff
@@ -1,27 +1,52 @@
 #!/usr/bin/env bash
 
-set -e
+set -e -o pipefail
+
+### Constants ###
+
+readonly TEST_DIR=projects/releases
 
 ### General Functions ###
 
 function exit_error() {
-    echo "--------- Captured output ---------"
-    echo "${output}"
-    echo "-----------------------------------"
     echo "ERROR: $*"
     exit 1
 }
 
-function print_test() {
-    echo "TEST: $*"
+function start_test() {
+    [[ -z "$CI" ]] || { printf '::endgroup::\n::group::%s\n' "$*" && return 0; }
+    local div=${*//?/=}
+    printf '%s\n%s\n%s\n' "$div" "$*" "$div"
+}
+
+# Copy over current git repo to testing directory to test any changes and avoid affecting the local repo.
+function prepare_test_repo() {
+    rm -rf "$TEST_DIR"
+    git -c advice.detachedHead=false clone . "$TEST_DIR"
```

Comment:
`advice.detachedHead=false` is necessary if youre in the middle of rebase and such